### PR TITLE
layers: Enable printf warnings for Log* methods

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
gcc and clang both can generate warnings about printf()
errors for any function if they are marked with:

__attribute__ ((format (printf, 4, 5)))

Change-Id: I91535aa12649f6a3a127f6141382a7598e198510

### CODE REVIEW

File: layers/best_practices_utils.cpp [hunk 1]
```diff
@@ -2181,7 +2181,7 @@ bool BestPractices::PreCallValidateCreateSampler(VkDevice device, const VkSample
                 "%s Creating a sampler object with wrapping modes which do not match (U = %u, V = %u, W = %u). "
                 "This may cause reduced performance even if only U (1D image) or U/V wrapping modes (2D "
                 "image) are actually used. If you need different wrapping modes, disregard this warning.",
-                VendorSpecificTag(kBPVendorArm));
+                VendorSpecificTag(kBPVendorArm), pCreateInfo->addressModeU, pCreateInfo->addressModeV, pCreateInfo->addressModeW);
```
Comment (ncesario-lunarg) [inline layers/best_practices_utils.cpp:2184]: Wow, seems like this could have ended badly without this.

**Review by ncesario-lunarg — APPROVED** [review]
LGTM!

**Review by Tony-LunarG — APPROVED** [review]
LGTM - How did some of this ever work?  Do you expected many validation error messagess to change?

**Comment by jeremyg-lunarg** [comment]
> LGTM - How did some of this ever work? Do you expected many validation error messagess to change?

I think some of them must not be hit very often because of how much stack garbage they'll print... 

A few messages might change, because of omitted function names or extra commas in the printf calls causing weird stuff to get printed.

**Pull request merged**